File: Euclid.Storage/Database/Access/NavigatorDao.cs
using Euclid.Storage.Database.Data;
using System.Collections.Generic;

namespace Euclid.Storage.Database.Access
{
    public class NavigatorDao
    {
        /// <summary>
        /// Get list of room categories
        /// </summary>
        public static List<NavigatorCategoryData> GetCategories()
        {
            using (var session = SessionFactoryBuilder.Instance.SessionFactory.OpenSession())
            {
                return session.QueryOver<NavigatorCategoryData>().List() as List<NavigatorCategoryData>;
            }
        }
    }
}

File: euclid-storage/src/main/java/euclid/storage/database/access/NavigatorDao.java
package euclid.storage.database.access;

import euclid.storage.database.SessionFactoryBuilder;
import euclid.storage.database.data.NavigatorCategoryData;
import org.hibernate.Session;
import org.hibernate.query.Query;

import java.util.List;

public class NavigatorDao {

    public static List<NavigatorCategoryData> getCategories() {
        try (Session session = SessionFactoryBuilder.getInstance().getSessionFactory().openSession()) {
            Query<NavigatorCategoryData> q = session.createQuery("FROM NavigatorCategoryData", NavigatorCategoryData.class);
            return q.getResultList();
        }
    }
}
